### ISSUE TITLE
Too many unnecessary formatting templates 

### ISSUE THREAD

**Issue by nuokh** [nextcloud/richdocuments#2754]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I open the formatting preset dropdown menue and get overhelmed by (what seems like) hundreds of text formatting presets.

![grafik](https://user-images.githubusercontent.com/9081831/212529905-8e675375-0092-4f19-8cbb-d3b3b81d6c05.png)

**Describe the solution you'd like**
I would like to see a maximum of maybe ten formatting templates. The default presets in [CKEditor](https://github.com/ckeditor/ckeditor5) are a good example of the amount of entries I expect when opening a formatting preset dropdown.

![grafik](https://user-images.githubusercontent.com/9081831/212529741-c1c832e9-37bc-4f59-98a6-4ccd78d40826.png)

**Additional context**
It came to our attention that while our IT department is fine with scrolling through the long list of templates, our end users are repulsed by NC Office as soon as they try to find a "normal" headline template.

**Comment by juliushaertl** [comment]
@jancborchardt Maybe we should carefully review that default list. Also I noticed that the list is different between the UI modes, so the tabbed menu only shows a subset of 8 styles:

<img width="454" alt="Screenshot 2023-01-16 at 10 11 15" src="https://user-images.githubusercontent.com/3404133/212640478-f95b3597-b5cd-47ea-a0fd-707a5b3c04a1.png">
<img width="135" alt="Screenshot 2023-01-16 at 10 10 54" src="https://user-images.githubusercontent.com/3404133/212640485-fcec13bf-ce31-4988-a869-ba9b9135bcde.png">

**Comment by nuokh** [comment]
> Did you by any chance create the document somewhere outside of Nextcloud?

No, quite the opposite. I created a new document on a fresh Nextcloud installation.

![image](https://user-images.githubusercontent.com/9081831/213989987-6e67f186-065b-4072-b051-8edb984476a9.png)

Then I pick the blank template.

![image](https://user-images.githubusercontent.com/9081831/213990102-c77ba47e-f708-44b9-9d57-91b52a7d3272.png)

Finally, I encounter the mentioned huge list of formatting presets in the new, blank document I created within Nextcloud.

![image](https://user-images.githubusercontent.com/9081831/213990366-2a20fa18-b821-4ce7-b39b-f020f513bb5e.png)

A few examples from the middle:

![image](https://user-images.githubusercontent.com/9081831/213990441-9dd13e20-20a5-4a14-81fc-0596082c1817.png)

**Comment by pedropintosilva** [comment]
Thanks @nuokh  for reporting I think we can divide the upcoming solution in 3 parts (as  discussed with @eszkadev @jancborchardt and @juliushaertl )
 - Lists (sidebar and toolbar ) should have the same items
 - The list should be sorted so the styles that are more common are up top; then styles that are in use but not common and then all the others
 - Later we could think on how to make this more context aware

**Comment by nuokh** [comment]
Sounds wonderful, thanks a lot!
